File: C_plus_plus/Remove_Duplicates_from_Sorted_Array_II.cpp
class Solution {
public:
    int removeDuplicates(int A[], int n) {
        if(n<3) return n;
        int current = A[0],count = 1,j=1;
        for(int i = 1 ; i < n ; ++i){
            if(A[i]==current){
                if(++count<3)
                    A[j++] = A[i];
            }
            else{
                A[j++] = A[i];
                current = A[i];
                count = 1;
            }
        }
        return j;
    }
};

File: java/Remove_Duplicates_from_Sorted_Array_II.java
// Pass the array only once
// Put the nums to the front of the array if the nums's count  doesn't  exceed 2.
// Time is O(n), space is O(1).

public class Solution {
    public int removeDuplicates(int[] nums) {
        int count = 1, len = 1;
        for (int i = 1; i < nums.length; ++i) {
            if (nums[i] == nums[i-1]) {
                if (++count <= 2)
                    nums[len++] = nums[i];
            }
            else {
                count = 1;
                nums[len++] = nums[i];
            }
        }
        return nums.length == 0 ? 0 : len;
    }
}
